pkg/build/build.go

```diff
@@ -350,6 +363,74 @@ func pullImage(ctx *buildContext, image string) (id string, err error) {
 	return pushTarget.Context().Digest(digest.String()).String(), nil
 }
 
+func isImageRemote(ctx *buildContext, image string) (bool, error) {
+	ref, err := images2.ParseReferenceNoDefault(image)
+	if err != nil {
+		return false, nil
+	}
+
+	_, err = remote.Index(ref, ctx.remoteOpts...)
+	if err != nil {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func resolveLocalImage(ctx *buildContext, image string) (string, error) {
+	// We need to be careful to make sure we only list ImageInstances in the namespace to which the user has access.
+	// We can determine the namespace based on the pushRepo, which is in the format <registry>/acorn/<namespace>.
+	// To be very safe, verify the structure of pushRepo before getting the namespace from it.
+	if len(strings.Split(ctx.pushRepo, "/")) != 3 || !strings.HasPrefix(ctx.pushRepo, "127.0.0.1:") || strings.Split(ctx.pushRepo, "/")[1] != "acorn" {
+		return "", fmt.Errorf("invalid pushRepo: %s", ctx.pushRepo)
+	}
+
+	namespace := strings.Split(ctx.pushRepo, "/")[2]
+
+	c, err := k8sclient.Default()
+	if err != nil {
+		return "", err
+	}
```

Comment:
Another thing. Initially I added the k8sclient to the `buildContext` struct and passed it down from the build server all the way to here. But it's pretty easy to just recreate it here. Not sure if there's an advantage to either approach, but I chose this one. Also curious to hear any thoughts on this.

Reply:
My opinion, for whatever it's worth, is that this is fine.